archivessnake/edit_notes.py

```diff
@@ -7,64 +7,67 @@
 import time
 
 from asnake.aspace import ASpace
+from asnake.utils import walk_tree
 #from fuzzywuzzy import fuzz
 from rapidfuzz import fuzz
 
-AS_REPOSITORY = 2
 OUTPUT_FILENAME = "out.csv"
 NOTE_TYPE_CHOICES = ["bioghist", "accessrestrict", "odd", "abstract", "arrangement", "userestrict", "fileplan", "acqinfo", "langmaterial", "physdesc", "prefercite", "processinfo", "relatedmaterial", "separatedmaterial"]
 ACTION_CHOICES = ["modify", "delete"]
 LEVEL = ["collection", "file", "series", "item", "all"]
 CONFIDENCE_RATIO = 97
-spreadsheet_path = os.path.join(os.path.abspath(os.path.dirname(__file__)), OUTPUT_FILENAME)
 
-def process_tree(args, resource):
+def process_tree(client, parser, writer):
     """Iterates through a given collection, file, or series provided by user input. Finds note content that matches user input and then deletes or modifies relevant notes according to user preference.  """
-    for record in resource.tree.walk:
+    args = parser.parse_args()
+    resource_id = client.get(f'/repositories/2/resources/{(args.resource_id)}').json()
+    for record in walk_tree(resource_id, client):
         updated = False
-        aojson = record.json()
-        if args.level in [record.level, "all"]:
-            notes = aojson.get("notes")
+        if args.level in [record['level'], "all"]:
+            notes = record['notes']
             for idx, note in reversed(list(enumerate(notes))):
-                if note["type"] == args.note_type:
-                    for subnote in note.get("subnotes"):
-                        content = subnote["content"]
+                if note['type'] == args.note_type:
+                    for subnote in note.get('subnotes'):
+                        content = subnote['content']
```

Comment:
```suggestion
                    for subnote in note.get('subnotes', []):
                        content = subnote['content']
```
As written, line 31 will error if there are no subnotes because the `get` method returns `None` by default if the key is missing. Adding a second argument returns that value instead, so I've added an empty list.